### ISSUE TITLE
Decoder Issue

### ISSUE THREAD

**Issue by robertalanm** [goldiusleonard/Dalle2_pytorch_project#1]
Hey, I have solved the 0.2-0.1 loss issue, you need to use the decoder trainer

```
decoder_trainer = DecoderTrainer(
    decoder,
    lr = 3e-4,
    wd = 1e-2,
    ema_beta = 0.99,
    ema_update_after_step = 1000,
    ema_update_every = 10,
)
```

just put this after your Decoder and you can start training :)

**Comment by goldiusleonard** [comment]
Thank you robert. Will try it outt :)

**Comment by robertalanm** [comment]
It looks like they have a diffusion piror trainer as well!! :)

```
diffusion_prior_trainer = DiffusionPriorTrainer(
    diffusion_prior,
    lr = 3e-4,
    wd = 1e-2,
    ema_beta = 0.99,
    ema_update_after_step = 1000,
    ema_update_every = 10,
)
```

**Comment by goldiusleonard** [comment]
I have check it out. Seems like its similar to mine. Have you test it out?

**Comment by robertalanm** [comment]
yes! I have generated something other than static! I have a decoder loss of 0.05 right now

**Comment by goldiusleonard** [comment]
> yes! I have generated something other than static! I have a decoder loss of 0.05 right now

thank you. The loss has go down gradually and not suddenly drop to 0.1-0.2. Your help really helps. Thanks

**Comment by goldiusleonard** [comment]
![image](https://user-images.githubusercontent.com/57057496/167284852-a58170e8-5189-4fd2-a80d-4d0eb65274c5.png)

I removed the optimizer and scheduler and added the diffusion trainer, but it doesnt make the loss go down. Do you have the same issue?

**Comment by robertalanm** [comment]
I have kept the optimizer and scheduler.

![image](https://user-images.githubusercontent.com/17856706/167439873-1278dae5-8975-4228-9774-9ee7315c8210.png)

I generated this from the model I trained over the weekend. I think DALLE2-pytorch needs more work, but it is definitely working.

The prior is not as good as the decoder right now
